lib/Service/YAPServer.hpp

```diff
@@ -0,0 +1,632 @@
+            isSent = send(channelNumber, payload, payloadSize);
+        }
+
+        return isSent;
+    }
+
+    /**
+     * Send a frame with the selected bytes.
+     * @param[in] channelName Channel to send frame to.
+     * @param[in] payload Byte buffer to be sent.
+     * @param[in] payloadSize Amount of bytes to send.
+     * @returns If payload succesfully sent, returns true. Otherwise, false.
+     */
+    bool sendData(const char* channelName, const uint8_t* payload, uint8_t payloadSize)
+    {
+        bool isSent = false;
+
+        if (nullptr != channelName)
+        {
+            isSent = sendData(getChannelNumber(channelName), payload, payloadSize);
+        }
+
+        return isSent;
+    }
+
+    /**
+     * Get Number of a channel by its name.
+     * @param[in] channelName Name of Channel
+     * @returns Number of the Channel, or 0 if not channel with the name is present.
+     */
+    uint8_t getChannelNumber(const char* channelName)
+    {
+        uint8_t idx = tMaxChannels;
+
+        if (nullptr != channelName)
+        {
+            for (idx = 0U; idx < tMaxChannels; idx++)
+            {
+                if (0U == strncmp(channelName, m_dataChannels[idx].m_name, CHANNEL_NAME_MAX_LEN))
+                {
+                    break;
+                }
+            }
+        }
+
+        return (idx == tMaxChannels) ? 0U : (idx + 1U);
+    }
+
+    /**
+     * Creates a new channel on the server.
+     * @param[in] channelName Name of the channel.
+     * It will not be checked if the name already exists.
+     * @param[in] dlc Length of the payload of this channel.
+     * @param[in] cb Callback for data incoming to this channel.
+     * @returns The channel number if succesfully created, or 0 if not able to create new channel.
+     */
+    uint8_t createChannel(const char* channelName, uint8_t dlc, ChannelCallback cb)
+    {
+        /* Using strnlen in case the name is not null-terminated. */
+        uint8_t nameLength = strnlen(channelName, CHANNEL_NAME_MAX_LEN);
+        uint8_t idx        = tMaxChannels;
+
+        if ((nullptr != channelName) && (0U != nameLength) && (MAX_DATA_LEN >= dlc) && (0U != dlc) && (nullptr != cb))
+        {
+            for (idx = 0U; idx < tMaxChannels; idx++)
+            {
+                if (nullptr == m_dataChannels[idx].m_callback)
+                {
+                    memcpy(m_dataChannels[idx].m_name, channelName, nameLength);
+                    m_dataChannels[idx].m_dlc      = dlc;
+                    m_dataChannels[idx].m_callback = cb;
+                    break;
+                }
+            }
+        }
+
+        return (idx == tMaxChannels) ? 0U : (idx + 1U);
+    }
+
+    /**
+     * Suscribe to a Channel to receive the incoming data.
+     * @param[in] channelName Name of the Channel to suscribe to.
+     * @param[in] callback Callback to return the incoming data.
+     */
+    void subscribeToChannel(const char* channelName, ChannelCallback callback)

```

Comment:
The Client cannot subscribe to more than one channel without being a minimum of 2 calls to the process() method. The following snippet will fail:

```
void setup()
{
    gYapClient.subscribeToChannel("TEST1", CB1);
    gYapClient.subscribeToChannel("TEST2", CB2);
}
```
This client would only try to subscribe to TEST2, as the m_pendingSuscribeChannel member would be overwriten on the second call to the method.

Currently I do not have a clean solution for this issue. (Should this be converted to a Github Issue?)

The first idea to come to mind is to use an array to hold a determined number of pending channels to suscribe to. However this raises other problems such as memory usage and how to track which channels are suscribed and which are still pending, and when/where does the suscribing process takes places.

Reply:
Yes, please raise an issue, because this it what everyone would expected to be able to.
The response might come never ... consider this too.